components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt

```diff
@@ -816,6 +816,16 @@ class PromptFeature private constructor(
         dialog.feature = this
 
         if (canShowThisPrompt(promptRequest)) {
+            // If the active prompt dialog has been updated, then dismiss it
+            activePrompt?.get()?.let {
+                if (!session.content.promptRequests
+                    .map { prompt -> prompt.uid }
+                    .contains(it.promptRequestUID)
+                ) {
+                    it.dismiss()
+                }
+            }
```

Comment:
We need to find a nicer way to achieve this 😅

Reply:
Another thing that we have to keep in mind is when we dismiss the dialog fragment, we could be dismissing the geckoView callbacks see

https://github.com/mozilla-mobile/android-components/blob/main/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/ChoiceDialogFragment.kt#L106-L130

Reply:
We can't use the `promptDialog.promptRequestUID == promptRequest.uid` comparison to check if the `activePrompt` was updated or not. This is because when we update the prompt request in `ChoicePromptUpdateDelegate`, [using](https://github.com/mozilla-mobile/android-components/blob/cbf866fe1fd61d16b3e4813f7c8bb7d26e6561b5/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/prompt/ChoicePromptUpdateDelegate.kt#L47) the `copy` function, the [`uid`](https://github.com/mozilla-mobile/android-components/blob/c2d8a5d17e5c9c3f517973c029e6299df7558a30/components/concept/engine/src/main/java/mozilla/components/concept/engine/prompt/PromptRequest.kt#L28) is re-generated. I reverted the changes back to this approach.